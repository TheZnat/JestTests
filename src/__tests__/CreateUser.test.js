import { createUser } from "../createUser";

describe("CreateUser", () => {
  it("should create a user with first name and last name only", () => {
    const user = createUser({ firstName: "John", lastName: "Doe" });
    expect(user).toMatchSnapshot();
  });

  it("should create a user with first name, last name and phone", () => {
    const user = createUser({
      firstName: "Anton",
      lastName: "Doe",
      phone: "1234567890",
    });
    expect(user).toMatchSnapshot();
  });

  it("should create a user with first name, last name and email", () => {
    const user = createUser({
      firstName: "John",
      lastName: "Doe",
      email: "john.doe@example.com",
    });
    expect(user).toMatchSnapshot(); // Переместите это внутрь блока it
  });
});
