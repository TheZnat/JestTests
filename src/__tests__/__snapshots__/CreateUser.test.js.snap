// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser should create a user with first name and last name only 1`] = `
{
  "email": "",
  "firstName": "John",
  "lastName": "Doe",
  "password": "password",
  "phone": "",
  "type": "USER",
}
`;

exports[`CreateUser should create a user with first name, last name and email 1`] = `
{
  "email": "john.doe@example.com",
  "firstName": "John",
  "lastName": "Doe",
  "password": "password",
  "phone": "",
  "type": "USER",
}
`;

exports[`CreateUser should create a user with first name, last name and phone 1`] = `
{
  "email": "",
  "firstName": "Anton",
  "lastName": "Doe",
  "password": "password",
  "phone": "1234567890",
  "type": "USER",
}
`;
